src/java/picard/analysis/directed/TargetMetricsCollector.java

```diff
@@ -301,17 +374,113 @@ public void setBaitSetName(final String name) {
+            // NB: ignores how to handle supplemental records when present for both ends by just using the mate information in the record.
+
+            if (!clipOverlappingReads || !rec.getReadPairedFlag()) return 0;
+
+            // Only clip records that are left-most in genomic order and overlapping.
+            if (rec.getMateAlignmentStart() < rec.getAlignmentStart()) return 0; // right-most, so ignore.  Do this first since getAlignmentEnd is expensive
+
+            final int numBasesToClip = rec.getAlignmentEnd() - rec.getMateAlignmentStart()  + 1;
+            if (numBasesToClip < 0) return 0; // left-most but not overlapping
+
+            return numBasesToClip;
+        }
+
+        private SAMRecord clipOverlappingBases(final SAMRecord rec, final int numOverlappingBasesToClip) {
+            // NB: ignores how to handle supplemental records when present for both ends by just using the mate information in the record.
+
+            if (!clipOverlappingReads || numOverlappingBasesToClip <= 0) return rec;
+
+            try {
+                // watch out for when the second read overlaps all of the first read
+                if (rec.getMateAlignmentStart() <= rec.getAlignmentStart()) { // make it unmapped
+                    if (noSideEffects) {
+                        final SAMRecord recClone = ((SAMRecord)rec.clone());
+                        recClone.setReadUnmappedFlag(true);
+                        return recClone;
+                    }
+                    else rec.setReadUnmappedFlag(true);
+                    return rec;
+                }
+
+                // clip it, clip it good
+                final Cigar cigar = CigarUtil.addSoftClippedBasesToEndsOfCigar(
+                        rec.getCigar(),
+                        false, // ignore, since we want 3' end genomic order
+                        numOverlappingBasesToClip, // want 3' end genomic order
+                        0
+                );
+                if (noSideEffects) {
+                    final SAMRecord recClone = ((SAMRecord)rec.clone());
+                    recClone.setCigar(cigar);
+                    return recClone;
+                }
+                else {
+                    rec.setCigar(cigar);
+                    return rec;
+                }
+            } catch (final CloneNotSupportedException e) {
+                throw new PicardException(e.getMessage(), e);
+            }
+        }
+
         /** Adds information about an individual SAMRecord to the statistics. */
-        public void acceptRecord(final SAMRecord rec) {
-            // Just plain avoid records that are marked as not-primary
-            if (rec.isSecondaryOrSupplementary()) return;
+        public void acceptRecord(final SAMRecord record) {
+            /*
+            Count metrics related to # of base and reads. Consider supplemental alignments for base counting but not record counting.
```

Comment:
Re-read the SAM spec 1.2: "Chimeric alignment" and "Multiple mapping".  Please verify my interpretation matches yours.

Supplementary alignments for a read are a set of linear alignments that do not have large overlaps.  Notice they *could* have overlaps, but likely they do not.  In the case of overlaps we are double-counting, but otherwise the prior implementation was under-counting.

Secondary alignments are when you want to represent a sub-optimal alignment, for example a less likely alignment to another location, say a ortholog or paralog.  These should neither be counted for counting reads nor counting bases.